### ISSUE TITLE
Publishing documentation failed

### ISSUE THREAD

**Issue by frankenjoe** [audeering/opensmile-python#52]
During the `2.3.0` building the documentation failed with a `requests.exceptions.HTTPError: 403 Client Error` raised by `audackend`. Building the documentation and when pushing the master worked fine. Maybe a random error?

cc @hagenw 

```
Running Sphinx v4.3.1
making output directory... done
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/numpy/objects.inv...
loading intersphinx inventory from https://pandas-docs.github.io/pandas-docs-travis/objects.inv...
intersphinx inventory has moved: https://docs.scipy.org/doc/numpy/objects.inv -> https://numpy.org/doc/stable/objects.inv
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 7 source files that are out of date
updating environment: [new config] 7 added, 0 changed, 0 removed
reading sources... [ 14%] api-smile
reading sources... [ 28%] changelog
reading sources... [ 42%] contributing
reading sources... [ 57%] genindex
reading sources... [ 71%] index
reading sources... [ 85%] install
reading sources... [100%] usage
executing usage

Extension error:
Cell raised uncaught exception:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_2788/2982938831.py in <module>
     10 
     11 
---> 12 db = audb.load(
     13     'emodb',
     14     version='1.1.1',

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audb/core/load.py in load(name, version, only_metadata, bit_depth, channels, format, mixdown, sampling_rate, tables, media, removed_media, full_path, cache_root, num_workers, verbose, **kwargs)
    786     if version is None:
    787         version = latest_version(name)
--> 788     deps = dependencies(name, version=version, cache_root=cache_root)
    789 
    790     # backward compatibility to audb<1.0.0

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audb/core/api.py in dependencies(name, version, cache_root)
    261     deps = Dependencies()
    262     if not os.path.exists(deps_path):
--> 263         backend = lookup_backend(name, version)
    264         with tempfile.TemporaryDirectory() as tmp_root:
    265             archive = backend.join(name, define.DB)

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audb/core/utils.py in lookup_backend(name, version)
     31 
     32     """
---> 33     return _lookup(name, version)[1]
     34 
     35 

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audb/core/utils.py in _lookup(name, version)
    123         header = backend.join(name, 'db.yaml')
    124 
--> 125         if backend.exists(header, version):
    126             return repository, backend
    127 

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audbackend/core/backend.py in exists(self, path, version, ext)
     96         """
     97         path = self.path(path, version, ext=ext)
---> 98         return self._exists(path)
     99 
    100     def get_archive(

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/audbackend/core/artifactory.py in _exists(self, path)
     59             # or
     60             # requests.exceptions.HTTPError: 403 Client Error
---> 61             return audfactory.path(path).exists()
     62         except self._non_existing_path_error:  # pragma: nocover
     63             return False

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/pathlib.py in exists(self)
   1286         """
   1287         try:
-> 1288             self.stat()
   1289         except OSError as e:
   1290             if not _ignore_error(e):

/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/site-packages/artifactory.py in stat(self, pathobj)
   1411         """
   1412         pathobj = pathobj or self
-> 1413         return self._accessor.stat(pathobj=pathobj)
   1414 
   1415     def with_name(self, name):

TypeError: stat() missing required argument 'path' (pos 1)
Exception ignored in: <function BaseEventLoop.__del__ at 0x7ff8b7f6c430>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/asyncio/base_events.py", line 682, in __del__
    self.close()
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/asyncio/unix_events.py", line 68, in close
    super().close()
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/asyncio/selector_events.py", line 92, in close
    self._close_self_pipe()
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/asyncio/selector_events.py", line 99, in _close_self_pipe
    self._remove_reader(self._ssock.fileno())
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/asyncio/selector_events.py", line 277, in _remove_reader
    key = self._selector.get_key(fd)
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/selectors.py", line 191, in get_key
    return mapping[fileobj]
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/selectors.py", line 72, in __getitem__
    fd = self._selector._fileobj_lookup(fileobj)
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/selectors.py", line 226, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
  File "/opt/hostedtoolcache/Python/3.10.1/x64/lib/python3.10/selectors.py", line 42, in _fileobj_to_fd
    raise ValueError("Invalid file descriptor: {}".format(fd))
ValueError: Invalid file descriptor: -1
Error: Process completed with exit code 2.
```

**Comment by frankenjoe** [comment]
During publishing `2.4.0` we encountered the same error again. So it seems not to be a random but consistent problem.

**Comment by frankenjoe** [comment]
Basically it looks like we cannot load `emodb` with `audb` any more.

**Comment by hagenw** [comment]
The problem is you are using Python 3.10 for publishing, but our code does not work with Python 3.10.
You need to choose a fixed version in `.github/workflows/publish.yml`, e.g.

```yaml
    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.8'
```

**Comment by frankenjoe** [comment]
@hagenw good observation. I created https://github.com/audeering/opensmile-python/pull/54

**Comment by frankenjoe** [comment]
Solved with #54
